From 1be17c96f5c43a256694e504a447821dd88782e2 Mon Sep 17 00:00:00 2001
From: Roman Dobrodii <rdobrodii@antmicro.com>
Date: Fri, 17 Mar 2023 20:13:21 +0100
Subject: [PATCH 4/8] nvidia/drivers/media: add
 nvidia,no-sensor-bus-introspection

Custom fix to prevent sensor_common stack from trying to be
clever and traversing parent I2C bus DT nodes.

Signed-off-by: Roman Dobrodii <rdobrodii@antmicro.com>
---
 nvidia/drivers/media/platform/tegra/camera/sensor_common.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/nvidia/drivers/media/platform/tegra/camera/sensor_common.c b/nvidia/drivers/media/platform/tegra/camera/sensor_common.c
index d50b22e4937b..107b9b0cefc7 100644
--- a/nvidia/drivers/media/platform/tegra/camera/sensor_common.c
+++ b/nvidia/drivers/media/platform/tegra/camera/sensor_common.c
@@ -757,6 +757,11 @@ static int sensor_common_init_device_config(
 	if (!np)
 		return -EINVAL;
 
+	if (of_property_read_bool(np, "nvidia,no-sensor-bus-introspection")) {
+		cfg->type = CAMERA_DEVICE_NONE;
+		return 0;
+	}
+
 	parent = of_get_parent(np);
 	if (!parent)
 		return -EINVAL;
-- 
2.25.1

